src/Bridge/Doctrine/MongoDbOdm/ItemDataProvider.php

```diff
@@ -0,0 +1,99 @@
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Doctrine\MongoDbOdm;
+
+use ApiPlatform\Core\Bridge\Doctrine\Common\Util\IdentifierManagerTrait;
+use ApiPlatform\Core\Bridge\Doctrine\MongoDbOdm\Extension\AggregationItemExtensionInterface;
+use ApiPlatform\Core\Bridge\Doctrine\MongoDbOdm\Extension\AggregationResultItemExtensionInterface;
+use ApiPlatform\Core\DataProvider\DenormalizedIdentifiersAwareItemDataProviderInterface;
+use ApiPlatform\Core\DataProvider\RestrictedDataProviderInterface;
+use ApiPlatform\Core\Exception\RuntimeException;
+use ApiPlatform\Core\Identifier\IdentifierConverterInterface;
+use ApiPlatform\Core\Metadata\Property\Factory\PropertyMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Property\Factory\PropertyNameCollectionFactoryInterface;
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\ODM\MongoDB\Aggregation\Builder;
+use Doctrine\ODM\MongoDB\DocumentManager;
+
+/**
+ * Item data provider for the Doctrine MongoDB ODM.
+ *
+ * @experimental
+ *
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ */
+final class ItemDataProvider implements DenormalizedIdentifiersAwareItemDataProviderInterface, RestrictedDataProviderInterface
+{
+    use IdentifierManagerTrait;
+
+    private $managerRegistry;
+    private $itemExtensions;
+
+    /**
+     * @param AggregationItemExtensionInterface[] $itemExtensions
+     */
+    public function __construct(ManagerRegistry $managerRegistry, PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory, iterable $itemExtensions = [])
+    {
+        $this->managerRegistry = $managerRegistry;
+        $this->propertyNameCollectionFactory = $propertyNameCollectionFactory;
+        $this->propertyMetadataFactory = $propertyMetadataFactory;
+        $this->itemExtensions = $itemExtensions;
+    }
+
+    public function supports(string $resourceClass, string $operationName = null, array $context = []): bool
+    {
+        return null !== $this->managerRegistry->getManagerForClass($resourceClass);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws RuntimeException
+     */
+    public function getItem(string $resourceClass, $id, string $operationName = null, array $context = [])
+    {
+        $manager = $this->managerRegistry->getManagerForClass($resourceClass);
+
+        if (!\is_array($id) && !($context[IdentifierConverterInterface::HAS_IDENTIFIER_CONVERTER] ?? false)) {
+            $id = $this->normalizeIdentifiers($id, $manager, $resourceClass);
+        }
+        $id = (array) $id;
+
+        $fetchData = $context['fetch_data'] ?? true;
+        if (!$fetchData && $manager instanceof DocumentManager) {
+            return $manager->getReference($resourceClass, reset($id));
+        }
+
+        $repository = $manager->getRepository($resourceClass);
+        if (!method_exists($repository, 'createAggregationBuilder')) {
```

Comment:
Seeing this check multiple times makes me think that we need to enforce a few more types when it comes to repositories in ODM, what do you think?